src/client/datascience/jupyter/kernels/cellExecution.ts

```diff
@@ -275,77 +273,115 @@ export class CellExecution {
     }
-
     /**
-     * Notify other parts of extension about the cell execution.
+     * Assume we run cell A
+     * Now run cell B, and this will update cell A.
+     * The way it works is, the request object created for cell A will get a message saying update your output.
+     * Cell A has completed, hence there's no execution task, we should create one or re-use an existing one.
+     * Creating one results in side effects such as execution order getting reset and timers starting.
+     * Hence where possible re-use an existing cell execution task associated with this document.
      */
-    private notifyCellExecution() {
-        const editor = this.editorProvider.editors.find((e) => e.file.toString() === this.cell.notebook.uri.toString());
-        if (!editor) {
-            throw new Error('No editor for Model');
+    private createTemporaryTask() {
+        if (this.cell.document.isClosed) {
+            return;
+        }
+        // If we have an active task, use that instead of creating a new task.
+        const existingTask = CellExecution.activeNotebookCellExecutionTask.get(this.cell.notebook);
+        if (existingTask) {
+            return existingTask;
+        }
+
+        // Create a temporary task.
+        this.previousResultsToRestore = { ...(this.cell.previousResult || {}) };
+        this.temporaryTask = notebook.createNotebookCellExecutionTask(
+            this.cell.notebook.uri,
+            this.cell.index,
+            this.kernelConnection.id
+        );
+        this.temporaryTask?.start({});
+        if (this.previousResultsToRestore.executionOrder && this.task) {
+            this.task.executionOrder = this.previousResultsToRestore.executionOrder;
+        }
+        return this.temporaryTask;
+    }
+    private endTemporaryTask() {
+        if (this.previousResultsToRestore?.executionOrder && this.task) {
+            this.task.executionOrder = this.previousResultsToRestore.executionOrder;
         }
-        if (editor && !(editor instanceof NotebookEditor)) {
-            throw new Error('Executing Notebook with another Editor');
+        if (this.previousResultsToRestore) {
+            this.temporaryTask?.end({
+                duration: this.previousResultsToRestore.duration,
+                success: this.previousResultsToRestore.success,
+                executionOrder: this.previousResultsToRestore.executionOrder
+            });
+        } else {
+            this.temporaryTask?.end({});
         }
-        editor.notifyExecution(this.cell);
+        this.previousResultsToRestore = undefined;
+        this.temporaryTask = undefined;
+    }
+
+    private async completedDueToCancellation() {
+        traceCellMessage(this.cell, 'Completed due to cancellation');
+        if (!this.cell.document.isClosed) {
+            await chainWithPendingUpdates(this.cell.notebook, (edit) => {
+                traceCellMessage(this.cell, 'Update cell statue as idle and message as empty');
```

Comment:
cell status as idle

Reply:
Damn, it, tried to sneak one past you...